product-service/src/types.ts

```diff
@@ -0,0 +1,12 @@
+import { ProxyResult } from 'aws-lambda';
+
+export interface IApiResponse extends ProxyResult {
+    headers: {
+        'Content-Type': string;
+        'Access-Control-Allow-Methods': string;
+        'Access-Control-Allow-Origin': string;
+    };
+    statusCode: number;
+    body: string;
+}
```

Comment:
You don't need to enrich this interface. ProxyResult is equal APIGatewayProxyResult and it already has these properties.
![image](https://user-images.githubusercontent.com/17197689/97974403-2ec50900-1dd0-11eb-99d6-024c9b6c4ef7.png)